### ISSUE TITLE
[Question] [Errno 21] Is a directory: '/root/autodl-tmp/chatgarment/datasets/'

### ISSUE THREAD

**Issue by liku0830** [biansy000/ChatGarment#17]
### Question

Dear Author,

Could you kindly provide the folder structure diagram of your project? I'm struggling to figure out the correct way to organize the dataset, which has resulted in many parts showing "None" in my implementation. Your guidance on the directory structure would be extremely helpful for resolving this issue.

Thank you very much for your time and support!
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   nebula_config ................ {
    "enabled": false, 
    "persistent_storage_path": null, 
    "persistent_time_interval": 100, 
    "num_of_version_in_retention": 2, 
    "enable_nebula_load": true, 
    "load_path": null
}
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   optimizer_legacy_fusion ...... False
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   optimizer_name ............... adamw
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   optimizer_params ............. {'lr': 0.0001, 'weight_decay': 0.0, 'betas': (0.9, 0.999)}
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   pipeline ..................... {'stages': 'auto', 'partition': 'best', 'seed_layers': False, 'activation_checkpoint_interval': 0, 'pipe_partitioned': True, 'grad_partitioned': True}
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   pld_enabled .................. False
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   pld_params ................... False
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   prescale_gradients ........... False
[2025-06-13 20:39:12,812] [INFO] [config.py:925:print]   scheduler_name ............... WarmupDecayLR
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   scheduler_params ............. {'total_num_steps': 20000, 'warmup_min_lr': 0, 'warmup_max_lr': 0.0001, 'warmup_num_steps': 100, 'warmup_type': 'linear'}
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   seq_parallel_communication_data_type  torch.float32
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   sparse_attention ............. None
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   sparse_gradients_enabled ..... False
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   steps_per_print .............. None
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   tensor_parallel_config ....... dtype=torch.float16 autotp_size=0 tp_overlap_comm=False tensor_parallel=TPConfig(tp_size=1, tp_grain_size=1, mpu=None, tp_group=None) injection_policy_tuple=None keep_module_on_host=False replace_with_kernel_inject=False
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   timers_config ................ enabled=True synchronized=True
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   train_batch_size ............. 32
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   train_micro_batch_size_per_gpu  4
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   use_data_before_expert_parallel_  False
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   use_node_local_storage ....... False
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   wall_clock_breakdown ......... False
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   weight_quantization_config ... None
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   world_size ................... 1
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   zero_allow_untested_optimizer  False
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   zero_config .................. stage=2 contiguous_gradients=True reduce_scatter=True reduce_bucket_size=500000000 use_multi_rank_bucket_allreduce=True allgather_partitions=True allgather_bucket_size=500000000 overlap_comm=True load_from_fp32_weights=True elastic_checkpoint=False offload_param=None offload_optimizer=None sub_group_size=1000000000 cpu_offload_param=None cpu_offload_use_pin_memory=None cpu_offload=None prefetch_bucket_size=50000000 param_persistence_threshold=100000 model_persistence_threshold=9223372036854775807 max_live_parameters=1000000000 max_reuse_distance=1000000000 gather_16bit_weights_on_model_save=False module_granularity_threshold=0 use_all_reduce_for_fetch_params=False stage3_gather_fp16_weights_on_model_save=False ignore_unused_parameters=True legacy_stage1=False round_robin_gradients=False zero_hpz_partition_size=1 zero_quantized_weights=False zero_quantized_nontrainable_weights=False zero_quantized_gradients=False zeropp_loco_param=None mics_shard_size=-1 mics_hierarchical_params_gather=False memory_efficient_linear=True pipeline_loading_checkpoint=False override_module_apply=True log_trace_cache_warnings=False
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   zero_enabled ................. True
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   zero_force_ds_cpu_optimizer .. True
[2025-06-13 20:39:12,813] [INFO] [config.py:925:print]   zero_optimization_stage ...... 2
[2025-06-13 20:39:12,813] [INFO] [config.py:911:print_user_config]   json = {
    "train_micro_batch_size_per_gpu": 4, 
    "gradient_accumulation_steps": 8, 
    "optimizer": {
        "type": "AdamW", 
        "params": {
            "lr": 0.0001, 
            "weight_decay": 0.0, 
            "betas": [0.9, 0.999]
        }
    }, 
    "scheduler": {
        "type": "WarmupDecayLR", 
        "params": {
            "total_num_steps": 2.000000e+04, 
            "warmup_min_lr": 0, 
            "warmup_max_lr": 0.0001, 
            "warmup_num_steps": 100, 
            "warmup_type": "linear"
        }
    }, 
    "fp16": {
        "enabled": false
    }, 
    "bf16": {
        "enabled": true
    }, 
    "gradient_clipping": 1.0, 
    "zero_optimization": {
        "stage": 2, 
        "contiguous_gradients": true, 
        "overlap_comm": true, 
        "reduce_scatter": true, 
        "reduce_bucket_size": 5.000000e+08, 
        "allgather_bucket_size": 5.000000e+08
    }
}
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:441: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
[2025-06-13 20:39:47,701] [INFO] [real_accelerator.py:254:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2025-06-13 20:39:49,134] [INFO] [logging.py:107:log_dist] [Rank -1] [TorchCheckpointEngine] Initialized with serialization = False
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:441: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
[2025-06-13 20:40:43,704] [INFO] [real_accelerator.py:254:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2025-06-13 20:40:45,130] [INFO] [logging.py:107:log_dist] [Rank -1] [TorchCheckpointEngine] Initialized with serialization = False
Start training ...
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:441: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
[2025-06-13 20:41:46,531] [INFO] [real_accelerator.py:254:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2025-06-13 20:41:47,929] [INFO] [logging.py:107:log_dist] [Rank -1] [TorchCheckpointEngine] Initialized with serialization = False
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:441: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
/root/miniconda3/lib/python3.12/site-packages/transformers/utils/generic.py:309: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  _torch_pytree._register_pytree_node(
[2025-06-13 20:42:43,442] [INFO] [real_accelerator.py:254:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2025-06-13 20:42:44,917] [INFO] [logging.py:107:log_dist] [Rank -1] [TorchCheckpointEngine] Initialized with serialization = False
